libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -56,66 +61,49 @@
   };
   private final Context context;
   private final String accessToken;
-  private final String userAgent;
-  private final OkHttpClient client;
+  private  final BlacklistClient blacklistClient;
+  private List<String> revokedKeys;
 
   CertificateBlacklist(Context context, String accessToken, String userAgent, OkHttpClient client) {
     this.context = context;
     this.accessToken = accessToken;
-    this.userAgent = userAgent;
-    this.client = client;
+    this.blacklistClient = new BlacklistClient(userAgent, client, this);
+    retrieveBlackList();
   }
 
-  List<String> retrieveBlackList() {
+  Boolean isBlacklisted(String hash) {
+    return revokedKeys.contains(hash);
+  }
+
+  private void retrieveBlackList() {
     File directory = context.getFilesDir();
-    List<String> blacklist = new ArrayList<>();
+    revokedKeys = new ArrayList<>();
 
     if (directory.isDirectory()) {
       File file = new File(directory, BLACKLIST_FILE);
 
       if (file.exists()) {
         try {
-          blacklist = obtainBlacklistContents(file);
-          blacklist.remove(BLACKLIST_HEAD);
-        } catch (IOException | IndexOutOfBoundsException exception) {
+          if (!isFileOpened(file)) {
```

Comment:
is there a lock free way to make sure we don't need to worry about at any time that file is read/written at the same time?

Reply:
yeah we can use boolean within the class to do the checking and use it to control read/write.